### PR TITLE
[DOC] Update the Glob docs with more details

### PR DESCRIPTION
**Pull request by mwichmann**
Covers xml file for manpage, very minor stuff in uguide, and docstring in code.

Also tweaked `Ignore` manpage entry, only because it was the very next description in Environment.xml and caught the eye.

Signed-off-by: Mats Wichmann <mats@linux.com>

## Contributor Checklist:

* [ ] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [X] I have updated `CHANGES.txt` (and read the `README.rst`)
* [X] I have updated the appropriate documentation

### CODE REVIEW

File: SCons/Environment.xml [hunk 1]
```diff
@@ -1901,123 +1901,148 @@ Nodes or strings representing path names.
...
+    <entry>matches everything</entry>
+  </row>
+  <row>
+      <entry><literal>?</literal></entry>
+      <entry>matches any single character</entry>
+  </row>
+  <row>
+      <entry><literal>[seq]</literal></entry>
+      <entry>matches any character in <emphasis>seq</emphasis>
+      (can be a list or a ranage).</entry>
```
Comment (bdbaddog) [inline SCons/Environment.xml]: typo ranage instead of range

File: SCons/Environment.xml [hunk 2]
```diff
@@ -1901,123 +1901,148 @@ Nodes or strings representing path names.
...
-to disable the search for matches on disk,
-thereby only returning matches among
-already-configured File or Dir Nodes.
-The default behavior is to
-return corresponding Nodes
-for any on-disk matches found.
+argument evaluates false,
+the search for matches on disk is disabled,
+and only matches from
+already-configured File or Dir Nodes are returned.
```
Comment (bdbaddog) [inline SCons/Environment.xml:1999]: I'd keep the explanation of the default (True) setting?

File: SCons/Environment.xml [hunk 3]
```diff
@@ -1901,123 +1901,148 @@ Nodes or strings representing path names.
...
-&f-VariantDir;,
-the returned Nodes should be from the
-corresponding source directory,
-not the local directory.
+argument evaluates true,
+and the local directory is a variant directory,
+then &f-Glob; returnes Nodes from
+the corresponding source directory,
+rather than the local directory.
+<!-- XXX what about generated files that don't exist in src but will be sources? -->
```
Comment (bdbaddog) [inline SCons/Environment.xml:2012]: Hrm. I hate those.. It find them in sources if they are there, and variantdir if they are there, and sources if they are in both.. I think.

File: SCons/Environment.xml (hunk 2 above)
Comment (mwichmann) [reply SCons/Environment.xml:1999]: Restored (in slighlly different form)

File: SCons/Environment.xml (hunk 1 above)
Comment (mwichmann) [reply SCons/Environment.xml]: fixed

**Pull request merged**